eng/pipelines/steps/set-base-image-override-options.yml

```diff
@@ -0,0 +1,24 @@
+parameters:
+  # Name of the pipeline variable to append the options to
+  variableName: null
+  dockerfileOs: ''
+  baseOverrideRegistry: ''
+
+steps:
+# Configures Image Builder options to override the base image tags for Ubuntu as defined in the Dockerfiles
+# and instead use the corresponding image from the Ubuntu ACR.
+- powershell: |
+    # For official builds, we configure the regex and substitution
+    # to use a different registry. (e.g. "ubuntu:20.04" becomes "ubuntu.azurecr.io/ubuntu:20.04")".
+    # These tags are multi-arch tags so they'll work across all the architectures we provide.
+    $baseOverrideRegex = "(${{ parameters.dockerfileOs }}:.+)"
```

Comment:
This will likely require normalizing the references of the base images or making the regex more flexible. For example, there exist tags like the following:

* `alpine:3.13`
* `arm32v7/alpine:3.13`
* `arm64v8/alpine:3.13`
* `mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.13`

The first 3 we want to replace but not the last. This regex doesn't correctly handle the architecture-specific image names. It would replace `arm32v7/alpine:3.13` with `arm32v7/<override-registry>alpine3.13` when it should just be `<override-registry>alpine3.13`.

In order to keep the regex as simple as we can, I think it would be best to update the Dockerfiles to use a standard convention in referencing base image tags for these Linux distros. For example, `arm32v7/alpine:3.13` should just be `alpine:3.13`. The `3.13` tag is multi-arch and the build infrastructure now specifies the `--platform` option in all relevant commands so we are assured in getting the correct architecture. But any refactoring of tags like this should be done in a separate PR.

Also note that for CentOS and Fedora, these are coming from their own respective registries (quay.io and registry.fedoraproject.org), so the regex will need to account for those cases as well.

Reply:
My intent here was to basically only match the first case, and ignore the last 3 (or more specifically, definitely the last one). I didn't realize the middle 2 were cases (but they aren't mirrored on the registry anyway). I'm happy to create a separate PR for the refactoring of the tags. If I want to ignore the final case, could I just change the regex to start with `^`? Then it wouldn't match at all?

Reply:
Will do some thinking about the fedora/centos cases. They are mirrored, but they're mirrored from Dockerhub. Is that correct for us?

Reply:
Ok, the centos images from quay.io aren't mirrored so only the Docker Hub images are relevant.

For Fedora, I'm not sure if the ones from Docker Hub are ok. Probably? Those were originally added in https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/483, so you could follow up with @wfurt on that.

Reply:
@wfurt, is there anything special about the fedora base images coming from fedoraproject vs dockerhub? If there isn't, I may put out a PR to remove the registry from the fedora images so that we can use the override registry work.

Reply:
What's the plan with this? I would expect that any normalization of base image tag usage in the Dockerfiles would be done before this PR gets merged.